### PR TITLE
Deboost Forte and enable C++17

### PR DESCRIPTION
**Pull request by fevangelista**
**Description**
This PR removes boost from Forte, transitions the CMake build system to use C++17, and applies clang-tidy to use nullptr instead of 0 or NULL

**Changes relevant to users**
- Needs a compiler compatible with C++17
- Requires a pre-compiled boost library

### CODE REVIEW

File: src/mrdsrg-spin-integrated/mrdsrg_smart_s.cc [hunk 1]
```diff
@@ -142,7 +142,7 @@ double MRDSRG::smart_s_davg_min_delta1() {
     }
     double davg_sum = std::accumulate(davg.begin(), davg.end(), 0.0);
     std::transform(davg.begin(), davg.end(), davg.begin(),
-                   std::bind1st(std::multiplies<double>(), 1.0 / davg_sum));
+                   [&](double x) { return x / davg_sum; });
```
Comment (fevangelista) [inline src/mrdsrg-spin-integrated/mrdsrg_smart_s.cc]: @lcyyork : can you check that this is the correct translation of the `bind1st` call here? `bind1st` is not supported in C++17 and a lambda should be the simplest solution.

Comment (lcyyork) [reply src/mrdsrg-spin-integrated/mrdsrg_smart_s.cc]: @fevangelista Thanks for the modification. This should be good.

Comment (fevangelista) [reply src/mrdsrg-spin-integrated/mrdsrg_smart_s.cc]: Great. If you have chance please review and comment/approve

Comment (lcyyork) [reply src/mrdsrg-spin-integrated/mrdsrg_smart_s.cc]: I do not have any problem so far.

**Pull request merged**